src/woo_search/search_index/index.py

```diff
@@ -48,12 +116,16 @@ class Document(ES_Document):
     registratiedatum: M[datetime] = mapped_field(Date(required=True))
     laatst_gewijzigd_datum: M[datetime] = mapped_field(Date(required=True))
 
+    # define attachment mapping without creating python field
+    attachment: M[Optional[Attachment]]
```

Comment:
In modern python, `Optional` is literally the equivalent of `T | None`

```suggestion
    attachment: M[Attachment | None]
```

I'm not sure I understand this construct through  - why is no `mapped_field(Attachment)` used? I would expect this to be similar to `publisher`.